docs/migrate/migrate-subgraph.md

```diff
@@ -0,0 +1,152 @@
+  .setBlockRange({ from: 6175243 })
+  .addLog('0x2E645469f354BB4F5c8a05B3b30A929361cf77eC', {
+    filter: [[
+      events['NewGravatar(uint256,address,string,string)'].topic,
+      events['UpdatedGravatar(uint256,address,string,string)'].topic,
+   ]],
+    data: {
+        evmLog: {
+            topics: true,
+            data: true,
+        },
+    } as const,
+});
+```
+
+In the snippet above we tell the squid processor to fetch logs emitted by the contract `0x2E645469f354BB4F5c8a05B3b30A929361cf77eC` that match of the specified topics. Note that the topic filter is a double array as required by the [selector specification](https://docs.ethers.io/v5/api/utils/abi/interface/#Interface--selectors). The configuration also specifies that the indexing should start from block `6175243` onwards (when the contract was deployed).
+
+### Transform and save the data
+
+The processor currently doesn't do much and simply outputs the data it fetches from the archive.
+
+Now we migrate the subgraph handlers which transform the event data into `Gravatar` objects. Instead of saving or updating gravatars one by one, `EvmBatchProcessor` receives an ordered batch of event items it is subscribed to (which can be inspected by the output). 
+
+We start with an auxiliary function that extracts the event data from the logs using the helper classes generated by `evm-codegen` from the ABI
+
+```ts
+function extractData(evmLog: any): { id: ethers.BigNumber, owner: string, displayName: string, imageUrl: string} {
+  if (evmLog.topics[0] === events['NewGravatar(uint256,address,string,string)'].topic) {
+    return events['NewGravatar(uint256,address,string,string)'].decode(evmLog)
+  }
+  if (evmLog.topics[0] === events['UpdatedGravatar(uint256,address,string,string)'].topic) {
+    return events['UpdatedGravatar(uint256,address,string,string)'].decode(evmLog)
+  }
+  throw new Error('Unsupported topic')
+}
+```
+
+Next, we processed with a batch handler collecting the updates from a single batch of EVM logs. The update is an ordered array of items which may span multiple blocks. Each item is either an `UpdateGravatar` event or a `NewGravatar` event, as was configured in the previous step
+
+```ts
+processor.run(new TypeormDatabase(), async (ctx) => {
+    const gravatars: Map<string, Gravatar> = new Map();
+    for (const c of ctx.blocks) {
+      for (const e of c.items) {
+        if(e.kind !== "evmLog") {
+          continue
+        }
+        const { id, owner, displayName, imageUrl } = extractData(e.evmLog)
+        gravatars.set(id.toHexString(), new Gravatar({
+          id: id.toHexString(),
+          owner: decodeHex(owner),
+          displayName,
+          imageUrl
+        })) 
+      }
+    }
+    await ctx.store.save([...gravatars.values()])
+});
+```
+
+The implementation is straightforward -- the newly created and/or updated gravatars are tracked by an in-memory map. The values are persistent in a single batch upsert via `cts.store.save(...)`.
```

Comment:
Maybe say "in-memory map, connecting the Gravatar's `id` with its related data", just to be more explicit.